design-proposals/kubebmc.md

```diff
@@ -0,0 +1,156 @@
+# Overview
+
+This proposal introduces [KubeBMC](https://github.com/starbops/kubebmc), a BMC emulator for Kubernetes that facilitates the deployment of software/applications/platforms such as OpenShift and OpenStack - whose installers typically require communication with bare-metal out-of-band management protocols like IPMI and Redfish - in KubeVirt VMs for development, testing, and debugging purposes, similar to the functionality provided by [VirtualBMC](https://opendev.org/openstack/virtualbmc) and [sushy-emulator](https://opendev.org/openstack/sushy-tools) but within a Kubernetes context.
+
+## Motivation
+
+Developers require the ability to deploy OpenShift and OpenStack clusters on local virtual environments. Existing solutions involve libvirt domains or QEMU VMs with BMC emulators, which are not directly compatible with KubeVirt due to its internal use of [`virtqemud`](https://libvirt.org/manpages/virtqemud.html), necessitating a Kubernetes-native solution. This was initially raised in the GitHub issue: [Expose a virtual BMC service per virtual machine, for IPMI and/or Redfish actions · Issue #9108 · kubevirt/kubevirt](https://github.com/kubevirt/kubevirt/issues/9108).
+
+## Goals
+
+- Provide essential BMC functionalities such as (rebooting, changing boot devices, and mounting virtual media) via IPMI and Redfish protocols against VMs backed by KubeVirt
+- Expose service endpoints for clients to access either in-cluster or externally
+- Enable deployment tools, e.g. [Metal3](https://metal3.io/), [Tinkerbell](https://tinkerbell.org/), [Bifrost](https://opendev.org/openstack/bifrost/), etc., to install software on VMs without adapting to the virtual environment by modifying their source code
+
+## Non Goals
+
+- Providing BMC functionalities for bare-metal machines
+
+## Definition of Users
+
+- Users who want to do out-of-band management to the VMs within a KubeVirt environment directly using bare-metal-like methods, e.g., IPMI or Redfish
+- Users who want to install software like OpenShift or OpenStack on VMs locally within a KubeVirt environment using existing deployment tools
+
+## User Stories
+
+- As a developer, I want to deploy a multi-node OpenShift cluster using its native installer in KubeVirt VMs.
+- As a tester, I need to quickly spin up and destroy OpenStack clusters in a virtual environment for testing purposes.
+
+## Repos
+
+No existing repos under the KubeVirt organization will be impacted. KubeBMC augments the KubeVirt ecosystem as an optional add-on providing an operator and the BMC implementation.
```

Comment:
I was under the impression you wanted to house the project under the KubeVirt org?

Reply:
IIUC this tries to express that the project can live under the KubeVirt umbrella but will not impact other repositories. It is purely optional, you can choose to deploy it on your KubeVirt cluster but you don't have to.